enhancements/storage/csi-ebs-operator.md

```diff
@@ -0,0 +1,149 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In the past storage drivers have been provided either in-tree or via external provisioners. With the move to advocating CSI as the way to deliver storage drivers, Red Hat needs to move the in-tree drivers (and others that our customers need) to use CSI. With the focus on operators in Red Hat OCP 4.x these CSI drivers should be installed in the form of an Operator.
+
+## Motivation
+
+* In-tree drivers will be removed from Kubernetes, we need to continue to make the drivers available and CSI is the way to do this.
+* AWS is a key cloud provider for OpenShift and use of EBS has been supported in both OpenShift 3.x and 4.x, we need to move this driver to be CSI, provided by an operator.
+* CSI drivers (incl. AWS EBS) will enable new storage features, not possible with in-tree volume plugins, like snapshots and volume cloning.
+
+### Goals
+
+* Create an operator to install the AWS EBS CSI driver.
+* Publish the operator on OperatorHub.
+* Package and ship downstream image of AWS EBS CSI driver (upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
+
+### Non-Goals
+
+* Driver creation (upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
+
+## Proposal
+
+OCP ships a new operator called aws-ebs-csi-driver-operator.
+
+1. The operator installs the AWS EBS CSI driver following the [recommended guidelines (currently work-in-progress)](https://github.com/openshift/enhancements/pull/139/files).
+2. The operator deploys the objects required by the AWS EBS CSI driver:
+   2.1 A ServiceAccount to be used the Controller Service of the CSI driver.
+   2.2 The RBAC rules to be used by the CSI driver.
+   2.3 The CSIDriver object representing the CSI driver.
+   2.4 A Deployment that runs the driver's Controller Service.
+   2.5 A DaemonSet that runs the driver's Node Service.
+   2.6 A non-default StorageClass that uses the CSI driver as a provisioner.
+3. The operator itself is installed by OLM once the user opts-in.
+4. The operator is deployed in the namespace `openshift-aws-ebs-csi-driver-operator`.
+   4.1 The CR is non-namespaced and is named `cluster`.
+5. The operator deploys the CSI driver in the namespace `openshift-aws-ebs-csi-driver`.
+6. The operator leverages the AWS credentials available in the `kube-system/aws-creds` secret.
+
+### API
+
+Given that no additional fields are needed, other than the standard ones, the following API should is provided by the operator:
+
+```go
+// EBSCSIDriver is a specification for a EBSCSIDriver resource
+type EBSCSIDriver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   EBSCSIDriverSpec   `json:"spec"`
+	Status EBSCSIDriverStatus `json:"status"`
+}
+
+// EBSCSIDriverSpec is the spec for a EBSCSIDriver resource
+type EBSCSIDriverSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+}
```

Comment:
We may have to add a field to control the number of controller replicas per zone